internal/tuf/tuf.go

```diff
@@ -196,7 +196,7 @@ func (d *Delegations) AddDelegation(delegation Delegation) {
 // Matches checks if any of the delegation's patterns match the target.
 func (d *Delegation) Matches(target string) bool {
 	for _, pattern := range d.Paths {
-		if ok, _ := path.Match(pattern, target); ok {
+		if ok, _ := filepath.Match(pattern, target); ok {
```

Comment:
this should certainly be path.Join as patterns will use `/`. We should likely make this clearer in the rule pattern docs. Also, #313 would change this anyway.